src/lexer/Lexer.ts

```diff
@@ -953,29 +966,36 @@ export class Lexer {
 
         let nextCharNeedsEscaped = false;
 
-        //finite loop to prevent infinite loop if something went wrong
-        for (let i = this.current; i < this.source.length; i++) {
+        //regexps can only occur when preceeded by exactly one of these tokens:
+        const previousKind = this.getPreviousNonWhitespaceToken()?.kind;
 
-            //if we reached the end of the regex, consume any flags
-            if (this.check('/') && !nextCharNeedsEscaped) {
-                this.advance();
-                //consume all flag-like chars (let the parser validate the actual values)
-                while (/[a-z]/i.exec(this.peek())) {
+        //preceeded by an allowed token, or if there are no previous tokens (i.e. this is the first token in the file).
+        if (PreceedingRegexTypes.has(previousKind) || !previousKind) {
```

Comment:
This is the core of the change. Check for only an approved token preceeding the opening slash.